c++/src/capnp/CMakeLists.txt

```diff
@@ -168,6 +168,7 @@ if(NOT CAPNP_LITE)
   set_target_properties(capnp_tool PROPERTIES CAPNP_INCLUDE_DIRECTORY
     $<JOIN:$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/..>,$<INSTALL_INTERFACE:${CMAKE_INSTALL_BINDIR}/..>>
   )
+  set_target_properties(capnp_tool PROPERTIES COMPILE_FLAGS -DCAPNP_INCLUDE_DIR="\\\"${CMAKE_INSTALL_FULL_INCLUDEDIR}\\\"")
```

Comment:
Any reason to use `COMPILE_FLAGS` over `COMPILE_DEFINITIONS`?
The latter seems more appropriate, and I think the escaping syntax will be lighter:

```cmake
set_target_properties(capnp_tool PROPERTIES
  COMPILE_DEFINITIONS CAPNP_INCLUDE_DIR=\"${CMAKE_INSTALL_FULL_INCLUDEDIR}\")
```

I don't know if the capnproto installation is relocatable today, but using `CMAKE_INSTALL_FULL_XXX` in the binary may harm relocatability of the package.
Unfortunately, I think the solution to this is more complex.
To do so, the binary should be able to locate it's own installation prefix at runtime, which is platform specific.

Reply:
Thank you for the hint, I didn't know about `COMPILE_DEFINITIONS`.
As for the relocatability, the installation was not relocatable before in the context of default include paths. Actually, the default include paths were hardcoded to `/usr/include` and `/usr/local/include`. The proposed change is not the ideal solution but is a bit more dynamic than before. The installation cannot be relocated but it can be at least installed locally. It was not possible before if using `cmake`.

Reply:
Yeah, this makes the cmake build match what the autotools build has done for a long time, which seems like an improvement.

Making the binary relocatable was discussed a bit in #1062, but it's not completely obvious what the right approach is.